File: MaximumProductOfWordLengths/MaximumProductofWordLengths.py
class Solution(object):
    def maxProduct(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        res = 0
        num = [0 for _ in range(len(words))]
        # a = 1 b = 10 c = 100 d = 1000
        for i in range(len(words)):
            for j in range(len(words[i])):
                num[i] = num[i] | (1 << (ord(words[i][j])) - 97)
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if num[i] & num[j] == 0 and len(words[i]) * len(words[j]) > res:
                    res = len(words[i]) * len(words[j])
        return res

File: MaximumProductOfWordLengths/MaximumProductofWordLengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) {
        int result = 0;
        vector<int> tmp(words.size(), 0);
        for(int i = 0;i != words.size();i++) {
        	for(auto j:words[i]) 
        	    tmp[i] |= 1 <<(j - 'a');
        }
        for(int i = 0; i < words.size(); i++) 
        	for(int j = i + 1; j < words.size(); j++) {
        		if(!(tmp[i] & tmp[j])) // && (words[i].length() * words[j].size() > result))
        			result = words[i].size() * words[j].size() > result ? words[i].size() * words[j].size():result;
        	}
        return result;
    }
};